src/entk_script/entk_script.py

```diff
@@ -0,0 +1,212 @@
+                        help='The queue from which resources are requested.')
+    parser.add_argument('walltime', type=int,
+                        help='The amount of time resources are requested in' + ' minutes')
+    parser.add_argument('cpus', type=int,
+                        help='Number of CPU Cores required to execute')
+    print (parser.parse_args())
+    return parser.parse_args()
+
+
+def generate_discover_pipeline(path, src_img):
+
+    '''
+    This function takes a single source image and path of set of images as an input on a specific
+    resource and returns a JSON file for all the images that exist in that path as an arguments.
+    '''
+    # Create a dicoverer Pipeline object
+    pipeline = Pipeline()
+    pipeline.name = 'Parser'
+
+    # Create a Stage object
+    stage = Stage()
+    stage.name = 'Parser-S0'
+    task = Task()
+    task.name = 'Parser-T0'
+    task.pre_exec = ['module load psc_path/1.1',
+                     'module load slurm/default',
+                     'module load intel/18.4',
+                     'module load xdusage/2.1-1',
+                     'module load python2/2.7.11_gcc_np1.11',
+                     'module load gcc/5.3.0',
+                     'module load opencv/2.4.13.2']
+    task.executable = 'python2'
+    # Assign arguments for the task executable
+    task.arguments = ['../img_parser.py', '%s' % path, '%s' % src_img]
+    task.download_output_data = ['images.json']
+    stage.add_tasks(task)
+    # Add Stage to the Pipeline
+    pipeline.add_stages(stage)
+
+    return pipeline
+
+
+def generate_pipeline(img1, img2, x1, y1, x2, y2, name):
+
+    """
+    This function will generate number of pipleines based on
+    the number of images being dicovered by the parser_pipleine.
+    """
+    args = args_parser()
+
+    # Create a Pipeline object
+    p = Pipeline()
+    p.name = name
+    source_img = img1
+    target_img = img2
+
+    # Create a "KeyPoints_Generation" Stage object
+    s1 = Stage()
+    s1.name = 'KeyPoints Generation'
+
+    # Create a Task object
+    t1 = Task()
+    t1.name = 'Task1'
+    t1.pre_exec = ['module load psc_path/1.1',
+                   'module load slurm/default',
+                   'module load intel/18.4',
+                   'module load xdusage/2.1-1',
+                   'module load python2/2.7.11_gcc_np1.11',
+                   'module load gcc/5.3.0',
+                   'module load opencv/2.4.13.2']
+    # Assign executable to the task
+    t1.executable = '../ASIFT/src/phase_2_keypoint_generation/fast_imas_IPOL/build/main'
+
+    # Assign arguments for the task executable
+    t1.arguments = ['-im1_gdal', source_img, 1000, 1000, x1, y1,
+                    '-im2_gdal', target_img, 1000, 1000, x2, y2,
+                    '-desc', args.desc]
+
+    t1.cpu_reqs = {'processes': 1,
+                   'threads_per_process': 1,
+                   'thread_type': None}
+    t1.download_output_data = ['data_matches.csv']
```

Comment:
I'm curious, why do you download this output? I see it is used from the next task and EnTK allows you to link data based on pipeline, stage and task place holders.

Wouldn't it make sense to use that? I'm afraid that when we scale this to multiple images this file will be overwritten in the client side. Using place holders will ensure that the next task gets the correct matches.

Reply:
This issue is resolved .